cmd/main.go

```diff
@@ -27,28 +30,45 @@ func main() {
 	config.SetOidcProvider()
 
 	// Get clientset for Authorizers
-	var clientset kubernetes.Interface
-	if config.EnableRBAC {
-		icc, err := rest.InClusterConfig()
-		if err != nil {
-			log.Fatalf("error getting in cluster configuration for RBAC client: %v", err)
-		}
-		clientset, err = kubernetes.NewForConfig(icc)
+	var clientset k8s.Interface
+	if config.EnableRBAC || config.EnableInClusterStorage {
+		var err error
+		clientset, err = kubernetes.GetClientSet()
 		if err != nil {
 			log.Fatalf("error getting kubernetes client: %v", err)
 		}
 	} else {
 		clientset = nil
 	}
```

Comment:
No need. `clientset` will be nil.
```suggestion
	}
```